src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13979,5 +13948,196 @@ int32[] arg2
                 } // end of class <>f__AnonymousDelegate0
                 """);
         }
+
+        [Fact]
+        public void ParamsArray_InferredConversion()
+        {
+            var source = """
+                var noParams = (int[] xs) => xs.Length;
+                var withParams = (params int[] xs) => xs.Length;
+                noParams = withParams; // 1
+                withParams = noParams; // 2
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (3,12): error CS0029: Cannot implicitly convert type '<anonymous delegate>' to 'System.Func<int[], int>'
+                // noParams = withParams; // 1
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "withParams").WithArguments("<anonymous delegate>", "System.Func<int[], int>").WithLocation(3, 12),
+                // (4,14): error CS0029: Cannot implicitly convert type 'System.Func<int[], int>' to '<anonymous delegate>'
+                // withParams = noParams; // 2
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "noParams").WithArguments("System.Func<int[], int>", "<anonymous delegate>").WithLocation(4, 14));
+        }
+
+        [Fact]
+        public void ParamsArray_MethodGroupConversion()
```

Comment:
nit: conversely, I think there's no method group conversion here, so the name seems odd.

Reply:
Thanks, changed test names: bf5ad716b7054554352703f15d08a4c7d79b42fa